File: StudyBuddy.BackendAngular/src/app/model/term.ts
import { Helper } from "../shared/helper";

export class Term {
    constructor(
        public id:string,
        public shortname:string,
        public name:string,
        public start:string,
        public end:string
    ) { }

    startAsDate():Date {
        return Helper.convertToDate(this.start);
    }

    endAsDate():Date {
        return Helper.convertToDate(this.end);
    }

    isPeriodValid():boolean {
        return this.endAsDate() > this.startAsDate();
    }

    isInPeriod(d:Date):boolean {
        return d >= this.startAsDate() && d <= this.endAsDate();
    }

    periodHasNoOverlapsWithOthers(objects:Term[]):boolean {
        for (let obj of objects)
            if (obj.isInPeriod(this.startAsDate()) || obj.isInPeriod(this.endAsDate()))
                return false;

        return true;
    }
}

File: StudyBuddy.Model/Term.cs
using System;

namespace StudyBuddy.Model
{
    public class Term : Entity
    {
        public string ShortName { get; set; }
        public string Name { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }

        public Term()
        {
            Start = DateTime.Now.Date;
            End = DateTime.Now.Date;
        }
    }
}